src/firebolt/async_db/util.py

```diff
@@ -12,14 +12,19 @@
     InterfaceError,
 )
 from firebolt.utils.urls import GATEWAY_HOST_BY_ACCOUNT_NAME
-from firebolt.utils.util import parse_url_and_params
+from firebolt.utils.util import UtilCache, parse_url_and_params
+
+_firebolt_async_system_engine_cache = UtilCache[Tuple[str, Dict[str, str]]]()
```

Comment:
It's global, since it's initialised on import. Or do you mean like a centralised one in, for example, connection.py? What would be an advantage of that?

Reply:
added common.cache